builder/simplify.cc

```diff
@@ -1309,6 +1309,8 @@ class simplifier : public node_mutator {
     return false;
   }
 
+  bool can_substitute_buffer(const depends_on_result& r) { return !(r.buffer_data() || r.var); }
```

Comment:
This should be static or a non-member

Reply:
Actually, maybe this should be in the `substitute.h` header. Anything that wants to use `substitute_buffer` would theoretically need to check this before trying to do so...